src/librustdoc/clean/mod.rs

```diff
@@ -2313,22 +2313,37 @@ impl Clean<Item> for (&hir::ForeignItem<'_>, Option<Ident>) {
     }
 }
 
-impl Clean<Item> for doctree::Macro {
+impl Clean<Item> for (&hir::MacroDef<'_>, Option<Ident>) {
     fn clean(&self, cx: &DocContext<'_>) -> Item {
-        Item::from_def_id_and_parts(
-            self.def_id,
-            Some(self.name.clean(cx)),
-            MacroItem(Macro {
-                source: format!(
-                    "macro_rules! {} {{\n{}}}",
-                    self.name,
-                    self.matchers
-                        .iter()
-                        .map(|span| { format!("    {} => {{ ... }};\n", span.to_src(cx)) })
-                        .collect::<String>()
-                ),
-                imported_from: self.imported_from.clean(cx),
-            }),
+        let (item, renamed) = self;
+        let name = renamed.unwrap_or(item.ident).name;
+        let tts = item.ast.body.inner_tokens().trees().collect::<Vec<_>>();
+        // Extract the spans of all matchers. They represent the "interface" of the macro.
+        let matchers = tts.chunks(4).map(|arm| arm[0].span()).collect::<Vec<_>>();
+        let source = if item.ast.macro_rules {
+            format!(
+                "macro_rules! {} {{\n{}}}",
+                name,
+                matchers
+                    .iter()
+                    .map(|span| { format!("    {} => {{ ... }};\n", span.to_src(cx)) })
+                    .collect::<String>(),
+            )
+        } else {
+            // This code currently assumes that there will only be one or zero matchers, as syntax
+            // for multiple is not currently defined.
```

Comment:
Multiple matchers is supported:

https://github.com/bjorn3/rustc_codegen_cranelift/blob/5988bbd24aa87732bfa1d111ba00bcdaa22c481a/src/intrinsics/mod.rs#L14-L27
```rust
macro intrinsic_pat {
    (_) => {
        _
    },
    ($name:ident) => {
        stringify!($name)
    },
    ($name:literal) => {
        stringify!($name)
    },
    ($x:ident . $($xs:tt).*) => {
        concat!(stringify!($x), ".", intrinsic_pat!($($xs).*))
    }
}
```

Reply:
Good catch, thanks!

Reply:
Thanks! I looked around for examples, but not hard enough apparently :P